### PR TITLE
More fixes for filtering empty paths in FossID scans

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt [hunk 1]
```diff
@@ -135,11 +135,11 @@ class FossId internal constructor(
          */
         private fun queryAuthenticator(repoUrl: String): String {
             val repoUri = repoUrl.toUri().getOrElse {
-                log.warn { "Host cannot be extracted for $repoUrl." }
+                log.warn { "The repository '$repoUrl' in not a valid URI." }
```
Comment (fviernau) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:138]: 'in not' -> 'is not'

Consider to reduce it to "The repository '$repoUrl' is not valid.

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt [hunk 2]
```diff
@@ -278,7 +278,7 @@ class FossId internal constructor(
                 }
 
             if (filteredPackages.isEmpty()) {
-                log.warn { "There is no package to scan !" }
+                log.warn { "There is no package to scan." }
```
Comment (fviernau) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:281]: Consider making this `.info`.

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt [hunk 3]
```diff
@@ -404,13 +409,13 @@ class FossId internal constructor(
...
         val scanCode = if (existingScan == null) {
-            log.info { "No scan found for $url and revision $revision. Creating scan ..." }
+            log.info { "No scan found for $url and revision $revision. Creating scan..." }
 
             val scanCode = namingProvider.createScanCode(projectName)
             val newUrl = if (config.addAuthenticationToUrl) queryAuthenticator(url) else url
             createScan(projectCode, scanCode, newUrl, revision)
 
-            log.info { "Initiating data download ..." }
+            log.info { "Initiating data download..." }
```
Comment (fviernau) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:418]: out of scope, maybe: "Initiating the download..."

**Review by fviernau — CHANGES_REQUESTED** [review]

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:138]: I went for yet another wording 😀

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt (hunk 2 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:281]: I prefer to keep this a warning as this case should be rather unusual (there should be at least the project package).

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt (hunk 3 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt]: Done.

**Review by fviernau — APPROVED** [review]

**Pull request merged**